corehq/apps/case_search/management/commands/stale_es_case_search_check.py

```diff
@@ -0,0 +1,288 @@
+
+    def add_arguments(self, parser):
+        parser.add_argument("--csv", action="store_true", default=False,
+            help="Write output as CSV data instead of padded table.")
+        parser.add_argument("-o", "--output", metavar="FILE", type=FileType("w"),
+            default=self.stdout, help="Write output to %(metavar)s (default=STDOUT).")
+        parser.add_argument("-q", "--query-size", metavar="COUNT", type=int, default=1000,
+            help="Limit queries to %(metavar)s case IDs per query (default=%(default)s)")
+        parser.add_argument("--mismatch-seconds", metavar="SECONDS", type=float,
+            default=(10 * 60),  # 10 minutes
+            help="Delta seconds for records to be considered a mismatch for " \
+            "(default=%(default)s)")
+        parser.add_argument("-d", "--divide-key", metavar="a[b]",
+            help="When running against all domains, only query domains whose "
+                 "first character of 'md5 hexdigest of name' is within range 'a->b'.")
+        parser.add_argument("-E", "--es-cases-cmp", action="store_true", default=False,
+            help="Compare case_search documents against ES cases index (instead of Couch/SQL)")
+        parser.add_argument("domains", metavar="DOMAIN", nargs="*",
+            help="Check timestamps for %(metavar)s")
+
+    def handle(self, domains, divide_key, mismatch_seconds, query_size, **options):
+        self.stderr.style_func = lambda x: x
+        logger = StubLogger(self.stderr)
+        # domains
+        all_cs_domains = sorted(fetch_all_case_search_domains())
+        if domains:
+            if divide_key:
+                raise CommandError("--divide-key option is mutually exclusive "
+                                   "with specified domains")
+            for index in range(len(domains) - 1, -1, -1):
+                if domains[index] not in all_cs_domains:
+                    logger.info("skipping domain %r (not in case_search index)",
+                                domains.pop(index))
+        else:
+            if divide_key:
+                try:
+                    domains = domain_subset(all_cs_domains, divide_key)
+                except ValueError as exc:
+                    raise CommandError(f"invalid divide-key: {exc!s}")
+                which = f"{len(domains)} of"
+            else:
+                domains = all_cs_domains
+                which = "all"
+            logger.info("checking %s %s case_search domains", which, len(all_cs_domains))
+
+        # warning condition if the `cases` document is newer than `case_search`
+        warn_case_newer_than = timedelta(seconds=mismatch_seconds)
+
+        table = Table(
+            header=["domain", "case_id", "delta", "cases_smo", "case_search_smo"],
+            max_col_width=36,
+        )
+        try:
+            for domain in domains:
+                logger.info("fetching cases for domain: %s", domain)
+                # query all cases and modified date
```

Comment:
nit: Rather than using comments to label sections, what do you think of splitting this (big) function into smaller functions with descriptive names?